tests/e2e/test_load.py

```diff
@@ -0,0 +1,77 @@
+import os
+import random
+import string
+
+import pytest
+import torch
+import ssl
+import functools
+import certifi
+from azstoragetorch.io import BlobIO
+from azure.identity import DefaultAzureCredential
+from azure.storage.blob import BlobServiceClient
+
+
+@pytest.fixture(scope="module")
+def model():
+    ssl._create_default_https_context = functools.partial(
+        ssl.create_default_context, cafile=certifi.where()
+    )
+    model = torch.hub.load("pytorch/vision:v0.10.0", "resnet101", pretrained=False)
```

Comment:
When we call out to `torch.load`, we should probably make sure to download it to the non-default hub directory for the following reasons:
* It possibly makes the test possibly not consistently reproducible if one environment has a cached version that differs from another 
* Tests generally should completely clean up after themselves and in this case we are leaving around cached models possibly
* In future CI environments, we may not have access to write in the home directory.

To address it, we should probably set one of the options [here](https://pytorch.org/docs/stable/hub.html#where-are-my-downloaded-models-saved) and set it to a temporary directory using the [`tmp_path`](https://docs.pytest.org/en/stable/reference/reference.html#std-fixture-tmp_path) fixture.